### ISSUE TITLE
Error message in summary.manyglm: length of 'dimnames' [1] not equal to array extent

### ISSUE THREAD

**Issue by skiyoshi** [dwarton/mvabund#23]
I get this error message when I use summary() following model fitting with manyglm. When p.uni="unadjusted" is not included, summary() works fine, but whenever p.uni= is included, error message appears. I am using R version 3.2.4 and just installed mvabund via install.packages function. If you could find out what I am doing wrong, that would be very helpful. Below are my data, code, and the error message. Thank you very much. 
Sincerely, 
Kiyoshi

mydata <- structure(list(Area = structure(1:12, .Label = c("CTV", "Falconbridge", 
"Franco", "Hwy144", "Kukagami", "Landfill", "Laurentian", "Nepewassi Lake", 
"ONeil", "Sand Bay Rd", "Tilton Lake", "Trout Lake"), class = "factor"), 
    Rank = structure(c(4L, 4L, 2L, 3L, 2L, 4L, 3L, 1L, 3L, 1L, 
    2L, 1L), .Label = c("Reference", "Moderate", "Semibarren", 
    "Barren"), class = "factor"), ANAM = c(1L, 4L, 0L, 1L, 19L, 
    1L, 4L, 88L, 9L, 41L, 6L, 21L), HYVE = c(10L, 7L, 12L, 2L, 
    3L, 7L, 6L, 26L, 4L, 9L, 5L, 11L), LICA = c(0L, 2L, 2L, 37L, 
    24L, 16L, 25L, 19L, 0L, 21L, 49L, 85L), LICL = c(71L, 32L, 
    52L, 33L, 77L, 45L, 40L, 110L, 91L, 80L, 56L, 93L), LIPI = c(16L, 
    61L, 69L, 14L, 4L, 34L, 51L, 96L, 52L, 38L, 11L, 20L), LISE = c(28L, 
    56L, 32L, 5L, 13L, 1L, 0L, 41L, 72L, 24L, 1L, 46L), LISY = c(0L, 
    4L, 13L, 2L, 9L, 11L, 1L, 72L, 0L, 14L, 5L, 6L), PSCR = c(7L, 
    13L, 18L, 9L, 16L, 13L, 9L, 36L, 9L, 17L, 13L, 29L)), .Names = c("Area", 
"Rank", "ANAM", "HYVE", "LICA", "LICL", "LIPI", "LISE", "LISY", 
"PSCR"), class = "data.frame", row.names = c(NA, -12L))

mv <- mvabund(mydata[,3:10])
fit <- manyglm(mv ~ Rank, data=mydata, family="negative.binomial")
summary(fit, resamp="monte.carlo", p.uni="unadjusted", test="LR")

Error in `rownames<-`(`*tmp*`, value = c("ANAM", "HYVE", "LICA", "LICL",  : 
  length of 'dimnames' [1] not equal to array extent

**Comment by aliceyiwang** [comment]
Hi, Kiyoshi,

I don't have this error by running your example with v3.11.7 from either
CRAN or GitHub. Can you please check if your local mvabund is updated to
the latest version?

Thanks,
Alice

**Comment by dwarton** [comment]
I can confirm this problem is in the 3.11.7 version of mvabund on CRAN, and that it has now been corrected in the files on github.  So if you install the latest version from github it should work for you, otherwise, you will have to wait for a 3.11.8 on CRAN...

**Comment by skiyoshi** [comment]
Thank you both for replying to my question. I will download it from GitHub. Sincerely, Kiyoshi

**Comment by MeggyC** [comment]
Hi there, sorry for digging this out. I am experiencing the same issue in version 4.1.3. 

The glm was built without problem but when asking for the summary I get this error:

code: 

```
summary(glm.vir, nBoot = 199,test = "LR") # summarise that glm
```

Error in dimnames(R) <- dimnames(X)[c(2, 2)]: length of 'dimnames' [1] not equal to array extent
Traceback:

1. summary(glm.vir, nBoot = 199, test = "LR")
2. summary.manyglm(glm.vir, nBoot = 199, test = "LR")

Thanks!

**Comment by dwarton** [comment]
Hi Meggy,
Can you confirm whether or not the code chunk Kiyoshi posted (above) works for you?  It worked for me on current versions of mvabund and R.  If that works, but your own data doesn't, then you have a new problem, in which case could you please send through some code to reproduce the issue?
thanks
David

**Comment by MeggyC** [comment]
Hi dwarton, I've tried Kiyoshi's code chunk and that works just fine. 

My code is as follows: 

```
clr_dfm <- read.csv("Z:/Accelerate Virus Projet/Virus biogeography/clr_dfm_t100.csv")


clr_dfm <- as.matrix(clr_dfm)
virdat <- mvabund(clr_dfm)
metadata <- read.csv("Z:/Accelerate Virus Projet/Virus biogeography/metadata_islands_no_strs.csv")
X <- metadata
glm.vir <- manyglm(virdat~., data=metadata)
summary(glm.vir, nBoot = 199,test = "LR") 
```
This returns the error

```
Error in dimnames(R) <- dimnames(X)[c(2, 2)]: length of 'dimnames' [1] not equal to array extent
Traceback:

1. summary(glm.vir, nBoot = 199, test = "LR")
2. summary.manyglm(glm.vir, nBoot = 199, test = "LR")
```

Please find my files attached ( I had to convert to xlsx for github)
[metadata_islands_no_strs.xlsx](https://github.com/aliceyiwang/mvabund/files/5215622/metadata_islands_no_strs.xlsx)
[clr_dfm_t100.xlsx](https://github.com/aliceyiwang/mvabund/files/5215626/clr_dfm_t100.xlsx)

**Comment by dwarton** [comment]
OK thanks for the details.  The main problem here seems to be that the model you are fitting has way too many terms in it.  There are only 22 rows of data so you don't have enough information there to fit more than a few predictors, but the model as specified had 20 predictors, many of which were completely redundant.  So for example this model fits fine:
```
glm.vir <- manyglm(virdat~Si_uM+NH4_uM, data=metadata)
summary(glm.vir, nBoot = 19,test = "LR") 

```
There were some bugs in the code that this problem found, to do with how manyglm handles reporting of coefficients when you fit a model with too many terms in it.  Those issues have been fixed in 4.1.5 on GitHub, thanks for bringing this to our attention.  But note that you really shouldn't be fitting this model in the first place!  You will need to simplify...
